docs/yaml/functions/summary.yaml

```diff
@@ -62,10 +62,30 @@ example: |
   ```
 
 posargs:
-  key:
-    type: str
-    description: The name of the new entry
+  key_or_dict:
+    type: str | dict
+    description: |
+      The name of the new entry, or a dictionary with values of the same type
+      as the `value` optarg.
 
+optargs:
   value:
     type: str | bool | int | dep | external_program | list[str | bool | int | dep | external_program]
-    description: The value to print for the `key`
+    description: |
+      The value to print for the `key`.  Only valid if `key_or_dict` is a str.
+
+kwargs:
+  bool_yn:
+    type: bool
+    default: false
+    description: Convert bool values to yes and no
+  section:
+    type: str
+    default: "''"
+    description: The section to put this summary information under.
```

Comment:
This feels a bit awkward to describe. Maybe it should t declare the default as `''` but say that if unspecified, this summary information goes into an unnamed leading section?